### DISCUSSION TITLE
HY2的Brutal算法疑问?

### DISCUSSION THREAD

**Discussion by tomxiang1** [apernet/hysteria#749] [Q&A]
如果客户端设定上下行100Mbps，服务端设定上下行500Mbps，假设当前网路的丢包率是50%，那么请问客户端与服务端之间会以什么速度发送udp包，是100Mbps固定速度发包，或者还是补偿到200Mbps的速度发包?

我的个人理解是，客户端设置的下行100Mbps是希望速度，服务端设置的上行500Mbps是丢包补偿最大速度，不知道这样理解是否正确?

**Comment by tobyxdd** [comment]
不正确，服务端速度的唯一作用是设置上限，比如：

客户端设置下载速度 100，服务端上传速度 200，最终是 100
客户端设置下载速度 100，服务端上传速度 10，最终是 10

Brutal 速度补偿的上限最高是 +25%，这个是代码写死的不会因为速度设置而更高

**Comment by tomxiang1** [comment]
> 不正确，服务端速度的唯一作用是设置上限，比如：
> 
> 客户端设置下载速度 100，服务端上传速度 200，最终是 100 客户端设置下载速度 100，服务端上传速度 10，最终是 10
> 
> Brutal 速度补偿的上限最高是 +25%，这个是代码写死的不会因为速度设置而更高

非常感谢解惑！

也就是说匹配速度为100Mbps时，理论客户端得到的最大下载（服务器上传）等于 100Mbps+25Mbps（丢包率>25%时），服务器最大发送速度125Mbps，客户端有效速度有可以达到100Mbps。

**Reply by tobyxdd** [reply]
基本正确，不过算错了，实际速度倍率=1/(1-丢包率)，丢包 20% 就会达到上限 125%
